api/v1alpha1/awsloadbalancercontroller_types.go

```diff
@@ -20,22 +20,85 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// +kubebuilder:validation:Enum=AWSShield;AWSAddonWAFv1;AWSAddonWAFv2
+type AWSAddon string
+
+const (
+	AWSAddonShield AWSAddon = "AWSShield"
+	AWSAddonWAFv1  AWSAddon = "AWSAddonWAFv1"
+	AWSAddonWAFv2  AWSAddon = "AWSAddonWAFv2"
+)
+
+// +kubebuilder:validation:Enum=Auto;Manual
+type SubnetTaggingPolicy string
+
+const (
+
+	// AutoSubnetTaggingPolicy enables automatic subnet tagging.
+	AutoSubnetTaggingPolicy SubnetTaggingPolicy = "Auto"
+
+	// ManualSubnetTaggingPolicy disables automatic subnet tagging.
+	ManualSubnetTaggingPolicy SubnetTaggingPolicy = "Manual"
+)
 
 // AWSLoadBalancerControllerSpec defines the desired state of AWSLoadBalancerController
 type AWSLoadBalancerControllerSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of AWSLoadBalancerController. Edit awsloadbalancercontroller_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// SubnetTagging describes how resource tagging will be done by the operator.
+	//
+	// When in "Auto", the operator will detect the subnets where the load balancers
+	// will be provisioned and have the required resource tags on them. Whereas when
+	// set to manual, this responsibility lies on the user.
+	//
+	SubnetTagging SubnetTaggingPolicy `json:"subnetTagging,omitempty"`
```

Comment:
`omitempty` implies that this is an optional field, let's add the optional comment markers then.

However don't we want to set a different defaulting strategy for this field?
Let's say we can default to `ManualSubnetTaggingPolicy`. This would make the API definition slightly more approachable for the end users.

Reply:
Added